Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp

```diff
@@ -137,6 +138,21 @@ void FiniteDifferenceSolver::MacroscopicEvolveM (
               M_xface(i, j, k, 2) += dt * (-PhysConst::mu0 * mag_gamma_interp) * ( M_xface(i, j, k, 0) * Hy_eff - M_xface(i, j, k, 1) * Hx_eff)
                 + dt * Gil_damp * ( M_xface(i, j, k, 0) * ( M_xface(i, j, k, 2) * Hx_eff - M_xface(i, j, k, 0) * Hz_eff)
                 - M_xface(i, j, k, 1) * ( M_xface(i, j, k, 1) * Hz_eff - M_xface(i, j, k, 2) * Hy_eff));
+
+          // temporary normalized magnitude of M_xface field at the fixed point
+              Real M_normalized = sqrt( pow(M_xface(i, j, k, 0),2.0) + pow(M_xface(i, j, k, 1),2.0) + pow(M_xface(i, j, k, 2),2.0) ) /
+                                  MacroscopicProperties::macro_avg_to_face(i,j,k,amrex::IntVect(1,0,0),mag_Ms_arr);
+
+              // check the normalized error
+              if (M_normalized < (1.0-M_normalized_error) || M_normalized > (1.0+M_normalized_error)){
+                  amrex::Print() << "i " << i << " j " << j << " k " << k << "\n";
+                  amrex::Abort("Exceed the normalized error of the M_xface field");
+              }
+
+              // normalize the M_xface field
+              M_xface(i,j,k,0) = M_xface(i,j,k,0)*M_normalized;
+              M_xface(i,j,k,1) = M_xface(i,j,k,1)*M_normalized;
+              M_xface(i,j,k,2) = M_xface(i,j,k,2)*M_normalized;
```

Comment:
shouldn't we *divide* by M_normalized?
Say mag_MS_arr is 1.0, but the magnitude of M is 1.1 (and thus M_normalized is 1.1); we need to *divide* M_xface by M_normalized.  (same comment for M_yface and M_zface)

Reply:
I think it should be multiplying M_norm. If M is 1.1 then there's a problem. M should never exceed Ms, so M/Ms should be smaller than one.

Reply:
So is |M|=0.9 and want it to be 1, we still want to multiply by M_normalized, right?
Should we outright abort then is |M|>1?

Reply:
Agreed! Sorry for the confusion in the document. I'll fix it. Andy is correct about the normalization

Reply:
I guess it should multiply M_xface by M_normalized. See step 3 of the section 5.1.1 in the EvolveM documentation. @jackieyao0114 can help to re-check this point.

Reply:
ok!